ebos/eclgenericoutputblackoilmodule.cc

```diff
@@ -633,6 +634,12 @@ setRestart(const data::Solution& sol,
         so -= sSol_[elemIdx];
     }
 
+    if (!rswSol_.empty()) {
+        if (sol.has("RSWSOL"))
+            rswSol_[elemIdx] = sol.data("RSWSOL")[globalDofIndex];
```

Comment:
I tried to build this but got an error.
[  1%] Building CXX object CMakeFiles/opmsimulators.dir/ebos/eclgenericoutputblackoilmodule.cc.o
/root/opm_beta/opm-simulators/ebos/eclgenericoutputblackoilmodule.cc: In member function ‘void Opm::EclGenericOutputBlackoilModule<FluidSystem, Scalar>::setRestart(const Opm::data::Solution&, unsigned int, unsigned int)’:
/root/opm_beta/opm-simulators/ebos/eclgenericoutputblackoilmodule.cc:658:40: error: no matching function for call to ‘Opm::data::Solution::data(const char [7]) const’
  658 |             rswSol_[elemIdx] = sol.data("RSWSOL")[globalDofIndex];
      |                                ~~~~~~~~^~~~~~~~~~
In file included from /root/opm_beta/opm-simulators/ebos/eclgenericoutputblackoilmodule.cc:31:
/root/opm_beta/opm-common/opm/output/data/Solution.hpp:51:25: note: candidate: ‘template<class T> std::vector<Scalar>& Opm::data::Solution::data(const string&)’
   51 |         std::vector<T>& data(const std::string& );
      |                         ^~~~
/root/opm_beta/opm-common/opm/output/data/Solution.hpp:51:25: note:   template argument deduction/substitution failed:
/root/opm_beta/opm-simulators/ebos/eclgenericoutputblackoilmodule.cc:658:40: note:   couldn’t deduce template parameter ‘T’
  658 |             rswSol_[elemIdx] = sol.data("RSWSOL")[globalDofIndex];
      |                                ~~~~~~~~^~~~~~~~~~
In file included from /root/opm_beta/opm-simulators/ebos/eclgenericoutputblackoilmodule.cc:31:
/root/opm_beta/opm-common/opm/output/data/Solution.hpp:54:31: note: candidate: ‘template<class T> const std::vector<Scalar>& Opm::data::Solution::data(const string&) const’
   54 |         const std::vector<T>& data(const std::string& ) const;
      |                               ^~~~
/root/opm_beta/opm-common/opm/output/data/Solution.hpp:54:31: note:   template argument deduction/substitution failed:
/root/opm_beta/opm-simulators/ebos/eclgenericoutputblackoilmodule.cc:658:40: note:   couldn’t deduce template parameter ‘T’
  658 |             rswSol_[elemIdx] = sol.data("RSWSOL")[globalDofIndex];
      |                                ~~~~~~~~^~~~~~~~~~
make[2]: *** [CMakeFiles/opmsimulators.dir/build.make:104: CMakeFiles/opmsimulators.dir/ebos/eclgenericoutputblackoilmodule.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:338: CMakeFiles/opmsimulators.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
root@MSI:~/opm_beta/opm-simulators/build# make
[  1%] Updating version information
[  1%] Built target update-version
Consolidate compiler generated dependencies of target opmsimulators
[  1%] Building CXX object CMakeFiles/opmsimulators.dir/ebos/eclgenericoutputblackoilmodule.cc.o
/root/opm_beta/opm-simulators/ebos/eclgenericoutputblackoilmodule.cc: In member function ‘void Opm::EclGenericOutputBlackoilModule<FluidSystem, Scalar>::setRestart(const Opm::data::Solution&, unsigned int, unsigned int)’:
/root/opm_beta/opm-simulators/ebos/eclgenericoutputblackoilmodule.cc:658:40: error: no matching function for call to ‘Opm::data::Solution::data(const char [7]) const’
  658 |             rswSol_[elemIdx] = sol.data("RSWSOL")[globalDofIndex];
      |                                ~~~~~~~~^~~~~~~~~~
In file included from /root/opm_beta/opm-simulators/ebos/eclgenericoutputblackoilmodule.cc:31:
/root/opm_beta/opm-common/opm/output/data/Solution.hpp:51:25: note: candidate: ‘template<class T> std::vector<Scalar>& Opm::data::Solution::data(const string&)’
   51 |         std::vector<T>& data(const std::string& );
      |                         ^~~~
/root/opm_beta/opm-common/opm/output/data/Solution.hpp:51:25: note:   template argument deduction/substitution failed:
/root/opm_beta/opm-simulators/ebos/eclgenericoutputblackoilmodule.cc:658:40: note:   couldn’t deduce template parameter ‘T’
  658 |             rswSol_[elemIdx] = sol.data("RSWSOL")[globalDofIndex];
      |                                ~~~~~~~~^~~~~~~~~~
In file included from /root/opm_beta/opm-simulators/ebos/eclgenericoutputblackoilmodule.cc:31:
/root/opm_beta/opm-common/opm/output/data/Solution.hpp:54:31: note: candidate: ‘template<class T> const std::vector<Scalar>& Opm::data::Solution::data(const string&) const’
   54 |         const std::vector<T>& data(const std::string& ) const;
      |                               ^~~~
/root/opm_beta/opm-common/opm/output/data/Solution.hpp:54:31: note:   template argument deduction/substitution failed:
/root/opm_beta/opm-simulators/ebos/eclgenericoutputblackoilmodule.cc:658:40: note:   couldn’t deduce template parameter ‘T’
  658 |             rswSol_[elemIdx] = sol.data("RSWSOL")[globalDofIndex];
      |                                ~~~~~~~~^~~~~~~~~~
make[2]: *** [CMakeFiles/opmsimulators.dir/build.make:104: CMakeFiles/opmsimulators.dir/ebos/eclgenericoutputblackoilmodule.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:338: CMakeFiles/opmsimulators.dir/all] Error 2
make: *** [Makefile:146: all] Error 2

So I changed it to
![image](https://github.com/OPM/opm-simulators/assets/77570362/ee9474f0-068b-4e07-916a-908e825e48ff)

and now it builds. Sorry for uploading an image but github removed the angle brackets everytime I commented.

Reply:
Thanks. I will update the PR accordingly. I will prioritize to get the PR into master to make it easier to use.

Reply:
for future reference: use html tags for angular brackets, ie sol.data & lt ;Scalar & gt; (without the spaces) or wrap it in a code block,
```
sol.data<Scalar>
```